folium/utilities.py

```diff
@@ -415,3 +415,11 @@ def _tmp_html(data):
     finally:
         if os.path.isfile(filepath):
             os.remove(filepath)
+
+
+def get_obj_in_upper_tree(element, cls):
```

Comment:
@jtbaker I reused that function from your Search plugin PR :)

Reply:
Nice! Good to see a spot to fit for it. :-)

One suggestion - maybe some type checking and a bombproof base case to avoid infinite recursion. Like:

```
from branca.element import Figure

...

def get_obj_in_upper_tree(element, cls):
    """Return the first object in the parent tree of class `cls`."""
    if not hasattr(element, '_parent'):
        raise ValueError('{} does not have a parent object.'.format(element))
    if isinstance(element, Figure) and cls is not Figure:
        raise ValueError('The top of the tree was reached without finding a {}'.format(cls))
    parent = element._parent
    if not isinstance(parent, cls):
        return get_obj_in_upper_tree(parent, cls)
    return parent
```

Reply:
Good one. I'll combine the two checks you suggested into one: when there's no more `_parent` the top of the tree was reached.

I don't think it will recurse infinitely because at some point there will be no more `_parent`. Maybe unless someone makes a cyclical graph. That would be interesting if someone manages to do that :) But then Pythons maximum recursion depth will kick in so at least an error will be thrown.